### PR TITLE
Revise `repr` overload as per request in #1988

### LINKED ISSUE #1988: Fix object `repr`s
In #1935, the color repr was changed from a knockoff tuple to <Color(...)> to be consistent with the vectors.

Sure, but the vectors are wrong too.

According to the repr docs:
> Return a string containing a printable representation of an object. For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to [eval()](https://docs.python.org/3/library/functions.html#eval); otherwise, the representation is a string enclosed in angle brackets that contains the name of the type of the object together with additional information often including the name and address of the object. A class can control what this function returns for its instances by defining a __repr__() method. If [sys.displayhook()](https://docs.python.org/3/library/sys.html#sys.displayhook) is not accessible, this function will raise [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError).

TLDR: If the class can make a representation that can be eval-ed, it shouldn't have angle brackets.

So it should be:
```
<Color(...)>  becomes Color(...)
<Vector2(...)> becomes Vector2(...)
<Vector3(...)> becomes Vector3(...)
<rect(...)> become Rect(...)
```

Perhaps other classes need modification too.

### PR DESCRIPTION
**Pull request by avaxar**
Closes #1988.

Revises the `repr` overload for `pygame.Color`, `pygame.Vector2`, `pygame.Vector3`, and `pygame.Rect` as @Starbuck5 has specified in #1988.

### CODE REVIEW

**Comment by Starbuck5** [comment]
Are there other classes that should be modified in this way? The list I came up with was not meant to be exhaustive.

**Comment by avaxar** [comment]
I don't think so. Perhaps `pygame.font.Font` could specify the font file? Though, I don't think it fits.

**Review by Starbuck5 — APPROVED** [review]

File: src_c/math.c [hunk 1]
```diff
@@ -1753,20 +1753,20 @@ static PyObject *
 vector_repr(pgVector *self)
 {
     /* The repr() of the largest possible Vector3 looks like
-     * "<Vector3({d}, {d}, {d})>" where 'd' has a maximum size of 32 bytes
+     * "Vector3({d}, {d}, {d})" where 'd' has a maximum size of 32 bytes
      *  so allocate a 16 + 3 * 32 == 112 byte buffer
```
[the remark this answers is not in the document]
Comment (avaxar) [reply src_c/math.c]: Done.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

Tested with this program:

```
from pygame import Color, Rect, Vector2

color_1 = Color(255, 128, 0)
print("color_1:", color_1)

colour_2 = eval(repr(color_1)).bgr
print("colour_2:", colour_2)

rect_1 = Rect(0, 0, 100, 100)
print("rect_1:", rect_1)

rect_2 = eval(repr(rect_1))
rect_2.topleft = (50, 75)
print("rect_2:", rect_2)

vec_1 = Vector2(20.0, 15.0)
print("vec_1:", vec_1)

vec_2 = eval(repr(vec_1)).normalize()
print("vec_2:", vec_2)
```
